File: Prototype/BigGp/src/Model/SmallNodes/Operators/Arithmetic/AdditiveOperator.java
package Model.SmallNodes.Operators.Arithmetic;

import Model.BaseNodes.ForStatement.ForIncrement;
import Model.Node;
import Model.SmallNodes.Expressions.Standart.TwoArgExpression;
import Model.SmallNodes.Operators.Operator;
import Serialization.Token;

import java.util.ArrayList;

public class AdditiveOperator extends Operator {

    @Override
    public void mutate() {
        if (this.getParentNode() instanceof ForIncrement){
            if (getValue().equals(" + ")){
                setOperator(" - ");
            } else {
                setOperator(" + ");
            }
        } else if (this.getParentNode() instanceof TwoArgExpression twoArgExpression){
            twoArgExpression.mutate();
        }
    }

    public AdditiveOperator(Node parentNode) {
        super(parentNode, "AdditiveOperator");
        if (getRandomInt(2) == 0){
            setOperator(" + ");
        } else {
            setOperator(" - ");
        }
    }

    public AdditiveOperator(Node parentNode, ArrayList<Token> tokens) {
        super(parentNode, tokens);
    }
}

File: Interactive.Genetic.Programming.Environment/src/Domain/Model/Nodes.Small/Operators.Arithmetic/AdditiveOperator.cs
using Model.Abstract;
using Model.Nodes.Big.For;
using Utils;

namespace Model.Nodes.Small.Operators.Arithmetic;

public sealed class AdditiveOperator : Operator
{
    public override void Mutate()
    {
        if (ParentNode is ForIncrement)
        {
            Value = Value switch
            {
                " + " => " - ",
                " - " => " + ",
                _ => throw new InvalidOperationException()
            };
        }
        else if (ParentNode is TwoArgExpression twoArgExpression)
        {
            twoArgExpression.Mutate();
        }
    }
    
    public AdditiveOperator(Node parentNode) : base(parentNode, "AdditiveOperator")
    {
        Value = RandomService.RandomPercentage() < 0.5
            ? " + "
            : " - ";
    }
    
    public AdditiveOperator(Node parentNode, IList<Token> tokens) : base(parentNode, tokens)
    {
    }
}